### PR TITLE
Login with twitter

### PR DESCRIPTION
**Pull request by AlaaTaima**
Relates #7

### CODE REVIEW

File: src/components/TwitterButton/index.js [hunk 1]
```diff
@@ -3,40 +3,38 @@ import propTypes from 'prop-types';
...
+        {
+          email: socialAuthUser.user.email,
+          name: socialAuthUser.user.displayName,
+          userID: socialAuthUser.user.uid,
+          goal: '',
+          createdByTwitter: true,
+        },
+        { merge: true }
+      );
+      await localStorage.setItem('userId', socialAuthUser.user.uid);
```
Comment (yosefanajjar) [inline src/components/TwitterButton/index.js]: We don't need to await here 
same goes for the line under it

File: src/components/TwitterButton/index.js [hunk 2]
```diff
@@ -3,40 +3,38 @@ import propTypes from 'prop-types';
...
+
+      await push(ROUTES.HOME);
+      this.setState({ error: null });
```
Comment (yosefanajjar) [inline src/components/TwitterButton/index.js:36]: Is it important to set the error state to null? cause I think we might not need to

File: src/components/TwitterButton/index.js [hunk 3]
```diff
@@ -46,8 +44,8 @@ class SignInTwitter extends Component {
       <Redirect to={ROUTES.HOME} />
```
Comment (yosefanajjar) [inline src/components/TwitterButton/index.js:46]: You can destructure the routes as well

File: src/components/TwitterButton/index.js [hunk 4]
```diff
@@ -46,8 +44,8 @@ class SignInTwitter extends Component {
...
     ) : (
       <button type="submit" className="facebook-btn" onClick={this.handleClick}>
-        <FacebookImg className="facebook-btn__img" />
-        <span className="facebook-btn__text">Facebook</span>
+        <TwitterImage className="facebook-btn__img" />
```
Comment (yosefanajjar) [inline src/components/TwitterButton/index.js]: Change the classes names 
```suggestion
        <TwitterImage className="twitter-btn__img" />
```

File: src/components/TwitterButton/index.js [hunk 5]
```diff
@@ -45,9 +43,9 @@ class SignInTwitter extends Component {
     return localStorage.getItem('userId') ? (
       <Redirect to={ROUTES.HOME} />
     ) : (
-      <button type="submit" className="facebook-btn" onClick={this.handleClick}>
-        <FacebookImg className="facebook-btn__img" />
-        <span className="facebook-btn__text">Facebook</span>
+      <button type="submit" className="twitter-btn" onClick={this.handleClick}>
+        <TwitterImage className="twitter-btn__img" />
+        <span className="twitter-btn__text">Twitter</span>
```
Comment (Fatmasiam) [inline src/components/TwitterButton/index.js]: move the error msg from the button

File: src/components/TwitterButton/index.js [hunk 6]
```diff
@@ -3,40 +3,40 @@ import propTypes from 'prop-types';
...
+          userID: socialAuthUser.user.uid,
+          goal: '',
+          createdByTwitter: true,
+        },
+        { merge: true }
+      );
+      localStorage.setItem('userId', socialAuthUser.user.uid);
+
+      push(ROUTES.HOME);
+      this.setState({ error: null });
```
Comment (Fatmasiam) [inline src/components/TwitterButton/index.js:36]: if we don't have errors there is no need to set the state to it 
but if you mean to empty the error message after making a new request you should put this line 
inline 14

File: src/containers/AccountSettings/accountSettings.jsx [hunk 7]
```diff
@@ -23,8 +22,11 @@ const accountSettings = ({ info, handleLogOut, loading }) => {
 
           <a
             href={
-              !info.createdByGoogle
+              // eslint-disable-next-line no-nested-ternary
```
Comment (Fatmasiam) [inline src/containers/AccountSettings/accountSettings.jsx]:
```suggestion
              // eslint-disable-next-line no-nested-ternary ???
```
**WHY**

File: src/containers/Firebase/firebase.js [hunk 8]
```diff
@@ -41,6 +41,10 @@ class Firebase {
   user = uid => this.db.doc(`users/${uid}`);
 
   users = () => this.db.collection('users');
+
+  journal = Jid => this.db.doc(`journals/${Jid}`);
```
Comment (Fatmasiam) [inline src/containers/Firebase/firebase.js]:
```suggestion
  journal = journalId => this.db.doc(`journals/${journalId}`);
```

**Comment by Fatmasiam** [comment]
and solve the conflict!

**Pull request merged**